spec/dry/types/schema_spec.rb

```diff
@@ -501,4 +501,45 @@ def self.meta
       specify { expect(type.transform_keys?).to be(true) }
     end
   end
+
+  describe '#merge' do
+    it 'adds keys and values from given schema' do
+      schema_1 = Dry::Types['hash'].schema(name: 'string')
+      schema_2 = Dry::Types['hash'].schema(age: 'integer')
+
+      schema = schema_1.merge(schema_2)
+
+      expect(schema.keys.map(&:name)).to eq([:name, :age])
+    end
+
+    it 'gives preference to given struct' do
+      schema_1 = Dry::Types['hash'].schema(foo: 'string')
+      schema_2 = Dry::Types['hash'].schema(foo: 'integer')
+
+      schema = schema_1.merge(schema_2)
+
+      expect(schema[foo: 1]).to eq(foo: 1)
+    end
+
+    it 'keeps type transformations from both schemas' do
+      transf_1 = ->(key) { key.constructor { 'Transf 1'} }
+      transf_2 = ->(key) { key.constructor { 'Transf 2'} }
+      schema_1 = Dry::Types['hash'].with_type_transform(transf_1).schema(foo: 'string')
+      schema_2 = Dry::Types['hash'].with_type_transform(transf_2).schema(bar: 'string')
+
+      schema = schema_1.merge(schema_2)
+
+      expect(schema[foo: 'foo', bar: 'bar']).to eq(foo: 'Transf 1', bar: 'Transf 2')
+    end
+
+    it 'preserves key transformations from the caller schema' do
+      schema_1 = Dry::Types['hash'].schema(foo: 'string').with_key_transform(&:to_sym)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use normalcase for variable numbers.](https://app.codacy.com/gh/dry-rb/dry-types/pullRequest?prid=4765184)